File: javaQuarkusShip/src/main/java/pl/battleships/javaquarkusship/api/InformationResource.java
package pl.battleships.javaquarkusship.api;

import pl.battleships.javaquarkusship.api.dto.InfoDto;

import java.util.List;

public class InformationResource implements InformationApi{
    @Override
    public InfoDto info() {
        return new InfoDto().name("javaQuarkus").details(
                List.of("java","quarkus")
        );
    }
}

File: kotlinQuarkusShip/src/main/kotlin/pl/battleships/kotlinquarkusship/resource/InformationResource.kt
package pl.battleships.kotlinquarkusship.resource

import pl.battleships.api.ApplicationInfoStackApi
import pl.battleships.api.dto.InfoDto
import javax.enterprise.context.ApplicationScoped

@ApplicationScoped
class InformationResource : ApplicationInfoStackApi {

    override fun info(): InfoDto {
        return InfoDto("kotlinQuarkus", listOf("kotlin", "quarkus"))
    }
}